test/Makefile

```diff
@@ -0,0 +1,92 @@
+
```

Comment:
Some files do not pass the linter at the moment. We shall fix this in the future. Fixing those files now would make the PR even bigger. We mainly wanted to have the framework in place for this PR.

Reply:
Do you want to add the linting without fixing the files? it would mean the travis (and all testing) fails for this branch (and develop when PR approved)

Reply:
I should point out that the linting fails here because /embrakes (3 or so) files weren't listed last year, so it wouldn't be too large of a fix, but it is work. That being said, I think many people would prefer the develop branch to always "work right out of the bag" when pulled.

Reply:
Honestly I'm not sure either what would be best, fixing the embrake files in this PR or just not including linting in the tests. I think for this PR, since it's so big already, it's fine if we don't include the linting so that it doesn't fail travis. 

In the future though, I think it's best if whenever a test/feature is added or the linter is changed, the PR with these changes should also make sure that the submitted code passes all tests and CI (this would mean getting other software members and having them submit commits to the PR as well). A stable `develop` branch is something we should aim for, as we should generally want to make sure it's releases ready at least most of the time.